src/main/java/com/casaculturaqxd/sgec/controller/dialog/DialogNovaInstituicao.java

```diff
@@ -0,0 +1,98 @@
+package com.casaculturaqxd.sgec.controller.dialog;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Optional;
+
+import org.controlsfx.control.textfield.AutoCompletionBinding;
+import org.controlsfx.control.textfield.TextFields;
+
+import com.casaculturaqxd.sgec.App;
+import com.casaculturaqxd.sgec.DAO.InstituicaoDAO;
+import com.casaculturaqxd.sgec.jdbc.DatabasePostgres;
+import com.casaculturaqxd.sgec.models.Instituicao;
+import com.casaculturaqxd.sgec.models.arquivo.ServiceFile;
+
+import javafx.fxml.FXMLLoader;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Alert.AlertType;
+import javafx.scene.control.ButtonBar;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.Dialog;
+import javafx.stage.FileChooser;
+import javafx.stage.FileChooser.ExtensionFilter;
+
+public class DialogNovaInstituicao extends Dialog<Instituicao> {
+  DatabasePostgres db = DatabasePostgres.getInstance("URL", "USER_NAME", "PASSWORD");
+  DialogInstituicaoController dialogInstituicaoController = new DialogInstituicaoController();
+  InstituicaoDAO instituicaoDAO = new InstituicaoDAO(db.getConnection());
+  Instituicao instituicao = null;
+  private Alert mensagem = new Alert(AlertType.NONE);
+
+  public DialogNovaInstituicao(ButtonType okButtonType) throws IOException{
+    super();
+    this.setTitle("Dialog Instituição");
+    ButtonType buttonTypeCancelar = new ButtonType("Cancelar", ButtonBar.ButtonData.CANCEL_CLOSE);
+    ButtonType buttonTypeCadastrar = new ButtonType("Cadastrar nova instituição", ButtonBar.ButtonData.APPLY);
+    FXMLLoader fxmlLoader = new FXMLLoader(App.class.getResource("view/Dialog/dialogInstituicoes.fxml"));
+    this.setDialogPane(fxmlLoader.load());
+    dialogInstituicaoController = fxmlLoader.getController();
+    this.getDialogPane().getButtonTypes().addAll(buttonTypeCancelar, buttonTypeCadastrar, okButtonType);
+    this.setResultConverter(dialogButton -> {
+      instituicao = dialogInstituicaoController.obterInstituicao();
+      if (dialogButton == okButtonType) {
+        if(instituicao != null){
+
+          if(!dialogInstituicaoController.getContribuicoes().getText().isEmpty()){
+            instituicao.setDescricaoContribuicao(dialogInstituicaoController.getContribuicoes().getText());
+          } else {
+            mensagem.setAlertType(AlertType.ERROR);
+            mensagem.setContentText("Não foi possivel realizar a vinculação: A descrição não pode ser vazia");
+            mensagem.show();
+            return null;
+          }
```

Comment:
### Manter o Dialog aberto caso a descrição ainda esteja vazia
Filtrar se a descrição foi inserida, só depois executar o retorno em setResultConverter
> A adição do eventFilter pode ser feita em um método separado e chamada em initialize

- [ ] remover essa verificação de setResultConverter
- [ ] fazer casting para botão: Button button = (Button) this.getDialogPane().lookupButton(okButtonType);
- [ ] executar a ação no bloco `if` no clique e consumir o evento
```java
button.addEventFilter(ActionEvent.ACTION, event -> {
     //funcao usada
    event.consume();    //consumir o evento impede a execução de chegar em getResult e fechar o Dialog
}